src/xrpld/app/tx/detail/SetTrust.cpp

```diff
@@ -182,6 +228,63 @@ SetTrust::preclaim(PreclaimContext const& ctx)
         }
     }
 
+    // Checking all freeze/deep freeze flag invariants.
+    auto sleRippleState =
+        ctx.view.read(keylet::line(id, uDstAccountID, currency));
+
+    if (sleRippleState)
+    {
+        if (ctx.view.rules().enabled(featureDeepFreeze))
+        {
+            bool const bNoFreeze = sle->isFlag(lsfNoFreeze);
+            bool const bSetFreeze = (uTxFlags & tfSetFreeze);
+            bool const bSetDeepFreeze = (uTxFlags & tfSetDeepFreeze);
+
+            if (bNoFreeze)
+            {
+                if (bSetFreeze || bSetDeepFreeze)
+                {
+                    // Cannot freeze the trust line if NoFreeze is set
+                    return tecNO_PERMISSION;
+                }
+            }
+            bool const bClearFreeze = (uTxFlags & tfClearFreeze);
+            bool const bClearDeepFreeze = (uTxFlags & tfClearDeepFreeze);
+            if ((bSetFreeze || bSetDeepFreeze) &&
+                (bClearFreeze || bClearDeepFreeze))
+            {
+                // Freezing and unfreezing in the same transaction should be
+                // illegal
+                return tecNO_PERMISSION;
+            }
+
+            bool const bHigh = id > uDstAccountID;
+            // Fetching current state of trust line
+            std::uint32_t uFlags = sleRippleState->getFieldU32(sfFlags);
```

Comment:
Fortunately, my previous comment is easy to resolve. Remove the check on `if (sleRippleState)` above, and change this line to use 0 if it's not set:
```suggestion
            std::uint32_t uFlags = sleRippleState ? sleRippleState->getFieldU32(sfFlags) : 0u;
```